### PR TITLE
Change catchup process to allow whitelisting of event sources

### PR DESCRIPTION
**Pull request by allanmckenzie**

### CODE REVIEW

File: event-store-management/event-store-management-core/src/main/java/uk/gov/justice/services/eventstore/management/catchup/process/EventSourceNameFilter.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private CatchupWhitelistedEventSourcesProvider catchupWhitelistedEventSourcesProvider;
+
+    /**
+     * Will return true if either the event-source name is found in our JNDI list of whitelisted event sources
+     * or if the JNDI value has not been set
+     *
+     * @param subscription The Subscription that contains the name of an event source used for catchup
+     * @return true if the event-source name is found in our JNDI list of whitelisted event sources
+     */
+    public boolean shouldRunCatchupAgainstEventSource(final Subscription subscription) {
```
Comment (Morbu) [inline event-store-management/event-store-management-core/src/main/java/uk/gov/justice/services/eventstore/management/catchup/process/EventSourceNameFilter.java:22]: is should the right naming convention for not test methods?

File: event-store-management/event-store-management-core/src/test/java/uk/gov/justice/services/eventstore/management/catchup/process/CatchupWhitelistedEventSourcesProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        when(initialContextFactory.create()).thenReturn(initialContext);
+        when(initialContext.lookup("java:global/catchup.event.source.whitelist")).thenThrow(namingException);
+
+        assertThat(catchupWhitelistedEventSourcesProvider.getWhiteListedEventSources(), is(empty()));
+
+        verify(logger).warn("No whitelisted catchup event-sources found: allowing catchup for all event-sources. Lookup made using JNDI name 'java:global/catchup.event.source.whitelist'.  Lookup error message: 'Name not found'");
+    }
+
+    @Test
+    public void shouldReturnEmptyListIfWhitelistedNamesIsSetToStar() throws Exception {
```
Comment (Morbu) [inline event-store-management/event-store-management-core/src/test/java/uk/gov/justice/services/eventstore/management/catchup/process/CatchupWhitelistedEventSourcesProviderTest.java:79]: test name, is no longer star

Comment (allanmckenzie) [reply event-store-management/event-store-management-core/src/test/java/uk/gov/justice/services/eventstore/management/catchup/process/CatchupWhitelistedEventSourcesProviderTest.java:79]: Good point

File: event-store-management/event-store-management-core/src/main/java/uk/gov/justice/services/eventstore/management/catchup/process/EventSourceNameFilter.java (hunk 1 above)
Comment (allanmckenzie) [reply event-store-management/event-store-management-core/src/main/java/uk/gov/justice/services/eventstore/management/catchup/process/EventSourceNameFilter.java:22]: I did think about that name a lot. The idea was to try to make the calling code understandable

 >  subscriptionsDescriptor.getSubscriptions()
                .stream()
                .filter(subscription -> eventSourceNameFilter.shouldRunCatchupAgainstEventSource(subscription))
                .sorted(priorityComparatorProvider.getSubscriptionComparator())
                
             
 Though if you can think of a better name let me know

File: event-store-management/event-store-management-core/src/test/java/uk/gov/justice/services/eventstore/management/catchup/process/CatchupWhitelistedEventSourcesProviderTest.java (hunk 2 above)
Comment (allanmckenzie) [reply event-store-management/event-store-management-core/src/test/java/uk/gov/justice/services/eventstore/management/catchup/process/CatchupWhitelistedEventSourcesProviderTest.java:79]: Fixed and pushed

**Pull request merged**